src/VersionParser.php

```diff
@@ -216,7 +216,7 @@ public function normalizeBranch($name)
      *
      * @return LinkConstraintInterface
      */
-    public function parseConstraints($constraints)
+    public static function parseConstraints($constraints)
```

Comment:
I would prefer keeping them normal methods. Static methods are not OOP. they are procedural code wrapped in a class.